src/fold/transformations/lags.py

```diff
@@ -54,10 +55,19 @@ def __init__(
 
     def transform(self, X: pd.DataFrame, in_sample: bool) -> pd.DataFrame:
         X = X.copy()
-        for column, lags in self.columns_and_lags:
-            lags = wrap_in_list(lags)
-            for lag in lags:
-                X[f"{column}_lag_{lag}"] = X[column].shift(lag)[-len(X) :]
+
+        if is_columns_all([self.columns_and_lags[0][0]]):
+            lags = wrap_in_list(self.columns_and_lags[0][1])
+            X = pd.concat(
+                [X]
+                + [X.shift(lag)[-len(X) :].add_suffix(f"_lag_{lag}") for lag in lags],
+                axis="columns",
+            )
+        else:
```

Comment:
there's a case when the user will want to apply lags to "all", but specific other lags to a different column.
right now, if "all" is present anywhere other than the first item, then we don't handle it at all.
but if other columns are present in the list, we ignore them.

if we want to keep this behaviour, could you add an exception at initalization time, that they either pass in a single "all" keyword, or they specify other columns than "all".

Reply:
I see, haven't thought of that. I'll implement it.